physical/oci_objectstorage/oci_objectstorage_ha.go

```diff
@@ -0,0 +1,554 @@
+
+	lockRecord, _, err := l.get(context.Background())
+	if err != nil {
+		return false, "", err
+	}
+	if lockRecord == nil {
+		return false, "", err
+	}
+	return true, string(lockRecord.Value), nil
+}
+
+// get retrieves the Value for the lock.
+func (l *Lock) get(ctx context.Context) (*LockRecord, string, error) {
+	l.backend.logger.Debug("Called getLockRecord()")
+
+	// Read lock Key
+
+	defer metrics.MeasureSince(metricGetHa, time.Now())
+	opcClientRequestId, err := uuid.GenerateUUID()
+	if err != nil {
+		l.backend.logger.Debug("physical/oci: getHa: error generating UUID")
+		opcClientRequestId = ""
+	}
+	l.backend.logger.Debug("physical/oci: getHa", "opc-client-request-id", opcClientRequestId)
+
+	request := objectstorage.GetObjectRequest{
+		NamespaceName:      &l.backend.namespaceName,
+		BucketName:         &l.backend.lockBucketName,
+		ObjectName:         &l.key,
+		OpcClientRequestId: &opcClientRequestId,
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, ObjectStorageCallsReadTimeout)
+	defer cancel()
+
+	response, err := l.backend.client.GetObject(ctx, request)
+	l.backend.logRequest("getHA", response.RawResponse, response.OpcClientRequestId, response.OpcRequestId, err)
+
+	if err != nil {
+		if response.RawResponse != nil && response.RawResponse.StatusCode == http.StatusNotFound {
+			return nil, "", nil
+		}
+
+		metrics.SetGauge(metricGetFailed, 1)
+		l.backend.logger.Error("physical/oci: Error calling GET", "err", err)
+		return nil, "", errwrap.Wrapf(fmt.Sprintf("failed to read Value for %q: {{err}}", l.key), err)
+	}
+
+	defer response.RawResponse.Body.Close()
+
+	body, err := ioutil.ReadAll(response.Content)
+	if err != nil {
+		metrics.SetGauge(metricGetFailed, 1)
+		l.backend.logger.Error("physical/oci: Error reading content", "err", err)
+		return nil, "", errwrap.Wrapf("failed to decode Value into bytes: {{err}}", err)
+	}
+
+	var lockRecord LockRecord
+	err = json.Unmarshal(body, &lockRecord)
+	if err != nil {
+		metrics.SetGauge(metricGetFailed, 1)
+		l.backend.logger.Error("physical/oci: Error un-marshalling content", "err", err)
+		return nil, "", errwrap.Wrapf(fmt.Sprintf("failed to read Value for %q: {{err}}", l.key), err)
+	}
+
+	return &lockRecord, *response.ETag, nil
+}
+
+func (l *Lock) writeLock() (bool, error) {
+	l.backend.logger.Debug("physical/oci: WriteLock() called")
+
+	// Create a transaction to read and the update (maybe)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// The transaction will be retried, and it could sit in a queue behind, say,
+	// the delete operation. To stop the transaction, we close the context when
+	// the associated stopCh is received.
+	go func() {
+		select {
+		case <-l.stopCh:
+			cancel()
+		case <-ctx.Done():
+		}
+	}()
+
+	lockRecordCache := loadLockRecordCache(l)
+	if (lockRecordCache == nil) || lockRecordCache.lockRecord == nil || lockRecordCache.lockRecord.Identity != l.identity || time.Now().UTC().Sub(lockRecordCache.time) > l.ttl {
```

Comment:
Split this across multiple lines for readability.

Reply:
`.UTC()` can be omitted

Reply:
We'd like to use store UTC on service side to standardize time for multi regional services.

Reply:
The calculation here is for duration, so the timezone operation has no effect (Go correctly tracks UTC internally regardless). Some examples of how .UTC() doesn't change the result: https://play.golang.org/p/N_D96Hs0ICi

Reply:
Yes, you're right, thanks for the details!